src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -126,6 +126,17 @@ public void addTask(Task task) {
         tasks.addTask(task);
     }
 
+    /**
+     * Replaces the given task {@code target} in the list with {@code editedTask}.
+     * {@code target} must exist in the task list.
+     * The task identity of {@code editedTask} must not be the same as another existing task in the task list.
+     */
+    public void setTask(Task target, Task editedTask) {
+        requireNonNull(editedTask);
```

Comment:
I was wondering if there be a check that the target Task should be NonNull as well